libcalico-go/lib/validator/v3/validator.go

```diff
@@ -1327,6 +1328,15 @@ func validateRule(structLevel validator.StructLevel) {
 			} else {
 				v4 = v4 || cidr.Version() == 4
 				v6 = v6 || cidr.Version() == 6
+
+				// Check for catch-all CIDR in negated context, which creates logical contradictions
+				if isNegatedField {
+					if (cidr.Version() == 4 && n == "0.0.0.0/0") ||
+						(cidr.Version() == 6 && n == "::/0") {
```

Comment:
If we parse the CIDR we can guarantee we are covering all possible representations of the IPv6 catch-all:

```suggestion
						(cidr.Version() == 6 && cidr.Mask.String() == cnet.MustParseCIDR("::/0").Mask.String() && cidr.IP.Equal(cnet.MustParseCIDR("::/0").IP)) {
```

Reply:
Thanks, I did just this!